### PR TITLE
Port storage handle tests to entity handles

### PR DESCRIPTION
**Pull request by jblebrun**
For now, the reference tests are commented out/ignored, since that functionality isn't implemented in the reference handles, yet.

### CODE REVIEW

File: javatests/arcs/core/entity/BUILD [hunk 1]
```diff
@@ -41,8 +42,14 @@ arcs_kt_jvm_library(
         ["*.kt"],
         exclude = TEST_SRCS,
     ),
+    visibility = ["//visibility:public"],
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/BUILD]: You shouldn't need this / this is a bad idea. This library should only be used inside this package. If it's supposed to be a more general purpose library, it should live in //java/.../testing or //java/.../testutil, and be marked `testonly = True`

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+        runBlockingTest { this.block() }
+    }
+
+    @Before
+    fun setup() {
+        RamDisk.clear()
+        DriverFactory.register(RamDiskDriverProvider())
+    }
+
+    @Suppress("UNCHECKED_CAST")
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt]: Just put this on the class

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    private suspend fun createWriteSingletonHandle(
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt]: Totally optional, but I'd suggest moving all the helper methods to the bottom of the file.

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+    ) = readHandleManager.createCollectionHandle(
+        HandleMode.ReadWrite,
+        name,
+        entitySpec,
+        storageKey
+    ) as ReadWriteCollectionHandle<Person>
+
+    @Test
+    fun singleton_writeAndReadBack() = testRunner {
+        val writeHandle =createWriteSingletonHandle()
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt]: Insert space

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+            updateDeferred.complete(it)
+        }
+        writeHandle.store(entity1)
+        assertThat(updateDeferred.await()).isEqualTo(entity1)
+    }
+
+    @Test
+    @Ignore("Fix when references are added in entity handles")
+    open fun singleton_referenceLiveness() = testRunner {
+        /*
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt]: How come it's both commented and ignored?

Comment (csilvestrini) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt]: Just read your PR description, ignore me! (or comment me out?)

File: javatests/arcs/core/entity/BUILD (hunk 1 above)
Comment (jblebrun) [reply javatests/arcs/core/entity/BUILD]: Thanks. I wasn't sure how to handle this, or why it wasn't visible. This file isn't test utilities, it's actual tests that get run in different configurations, so putting it in `//java` doesn't feel quite right.

File: javatests/arcs/core/entity/HandleManagerTestBase.kt (hunk 3 above)
Comment (jblebrun) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt]: oh yes, I meant to do that!

**Comment by jblebrun** [comment]
I think those tests should probably be ported into this group of tests, as well.

**Pull request merged**